### DISCUSSION TITLE
Explain how to use Benthos with Pub/Sub like I am 5 ...

### DISCUSSION THREAD

**Discussion by michealroberts** [redpanda-data/connect#2457] [Q&A]
So, I am a certified moron, let alone I have briefly skim-read the documentation for Benthos.

What I would like to do, is every time a new file gets pushed (uploaded) to a Google cloud storage bucket (Firebase storage technically), I run some "code" using the Blob file as the starting point ...

I would essentially like to apply some pre-processing to the raw image data that lands in the bucket.

Is someone able to explain to me how I do this with Benthos, or if it is even possible?

Many thanks 🙏

**Comment by mihaitodor** [comment]
Hi @michealroberts 👋 You're probably looking for the enhancement described in #1658 which aims to add PubSub support for the Google Cloud Storage input. Unfortunately, that code is still a bit rough and I don't know when it will get prioritised. Until that gets merged, one possibility would be to write your own custom input [plugin](https://github.com/benthosdev/benthos-plugin-example). Otherwise, maybe you can use Benthos as a [Cloud Function](https://cloud.google.com/functions/docs/calling) and have it run on certain triggers, but not sure if that just works out of the box.
